### PR TITLE
add profile for NAH.SH

### PR DESCRIPTION
**Pull request by juliuste**

### CODE REVIEW

File: p/nahsh/index.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+}
+
+const parseLocation = (profile, l, lines) => {
+	const res = _parseLocation(profile, l, lines)
+	// weird fix for empty lines, e.g. IC/EC at Flensburg Hbf
+	if(res.lines){
+		res.lines = res.lines.filter(x => x.id && x.name)
+	}
+
+	// remove trailing zeroes, todo
```
Comment (derhuerst) [inline p/nahsh/index.js]: what does "todo" mean here?

File: p/nahsh/index.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	body.auth = {aid: 'r0Ot9FLFNAFxijLW'}
+	body.lang = 'de'
+
+	return body
+}
+
+const parseLocation = (profile, l, lines) => {
+	const res = _parseLocation(profile, l, lines)
+	// weird fix for empty lines, e.g. IC/EC at Flensburg Hbf
+	if(res.lines){
```
Comment (derhuerst) [inline p/nahsh/index.js]: code style

File: p/nahsh/index.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+		res.lines = res.lines.filter(x => x.id && x.name)
+	}
+
+	// remove trailing zeroes, todo
+	if(res.id && res.id.length > 2){
+		while(res.id.slice(0, 1) === '0'){
```
Comment (derhuerst) [inline p/nahsh/index.js]: why not `res.id.replace(/^0+/, '')`?

File: p/nahsh/index.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+		}
+	}
+
+	return res
+}
+
+const createParseLine = (profile, operators) => {
+	const parseLine = _createParseLine(profile, operators)
+
+	const parseLineWithMode = (l) => {
```
Comment (derhuerst) [inline p/nahsh/index.js]: This will be obsolete when merging from `master`.

File: p/nahsh/index.js [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+		const res = parseJourney(j)
+
+		if (
+			j.trfRes &&
+			Array.isArray(j.trfRes.fareSetL) &&
+			j.trfRes.fareSetL.length > 0
+		) {
+			res.tickets = []
+
+			for(let t of j.trfRes.fareSetL){
```
Comment (derhuerst) [inline p/nahsh/index.js]: code style

File: p/nahsh/index.js [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+				const tariff = t.desc
+				if(!tariff || !Array.isArray(t.fareL)) continue
```
Comment (derhuerst) [inline p/nahsh/index.js]: code style

File: p/nahsh/index.js [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+			}
+		}
+
+		return res
+	}
+
+	return parseJourneyWithTickets
+}
+
+const defaultProducts = {
```
Comment (derhuerst) [inline p/nahsh/index.js]: This will be obsolete when merging from `master`.

File: p/nahsh/index.js [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+
+const nahshProfile = {
+	locale: 'de-DE',
+	timezone: 'Europe/Berlin',
+	endpoint: 'http://nah.sh.hafas.de/bin/mgate.exe',
+	transformReqBody,
+
+	products: products.allProducts,
+
+	parseProducts: createParseBitmask(products.allProducts, defaultProducts),
```
Comment (derhuerst) [inline p/nahsh/index.js]: This will be obsolete when merging from `master`.

File: p/nahsh/index.js (hunk 2 above)
Comment (juliuste) [reply p/nahsh/index.js]: could you be a bit more specific? 😄

Comment (derhuerst) [reply p/nahsh/index.js]: `if (res.lines) {`

File: p/nahsh/index.js (hunk 5 above)
Comment (derhuerst) [reply p/nahsh/index.js]: `for (...) {`

File: p/nahsh/index.js (hunk 6 above)
Comment (derhuerst) [reply p/nahsh/index.js]: `if (...) {`

File: p/nahsh/index.js (hunk 2 above)
Comment (juliuste) [reply p/nahsh/index.js]: Ah ok, will fix

**Comment by derhuerst** [comment]
Except the small stylistic notes, I have nothing to complain about. 😛

Will try to solve the outstanding todos before merging.

**Comment by derhuerst** [comment]
Created #34 to keep track of the missing station names and skipped the test for now.

File: p/nahsh/index.js (hunk 1 above)
Comment (derhuerst) [reply p/nahsh/index.js]: the comment says "trailing", while the code removes leading zeros

File: parse/products-bitmask.js [hunk 9]
```diff
@@ -21,6 +21,9 @@ const createParseBitmask = (allProducts, defaultProducts) => {
 				res[product.product] = true
 				bitmask -= product.bitmask
 			}
```
Comment (derhuerst) [inline parse/products-bitmask.js]: `} else {`

**Comment by derhuerst** [comment]
LGTM

**Pull request merged**